src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VsCredentialProviderAdapter.cs

```diff
@@ -46,13 +48,27 @@ public VsCredentialProviderAdapter(IVsCredentialProvider provider)
             }
 
             // TODO: Extend the IVS API surface area to pass down the credential request type.
-            var credentials = await _provider.GetCredentialsAsync(
+
+            // Telling the credential provider to cancel the request in situations like PM UI being closed or the search query changing
+            // is disruptive, because it may prompt the customer for interactive input, but then not get/cache a token, so the next request
+            // has to prompt the customer again. Therefore, let's only tell the provider to cancel when VS is shutting down, so it has
+            // the opportunity to cache tokens and not need to prompt the customer in the near future.
+            Task<ICredentials> task = _provider.GetCredentialsAsync(
                 uri,
                 proxy,
                 isProxyRequest: type == CredentialRequestType.Proxy,
                 isRetry: isRetry,
                 nonInteractive: nonInteractive,
-                cancellationToken: cancellationToken);
+                cancellationToken: VsShellUtilities.ShutdownToken);
```

Comment:
My gut feeling is that solution close would be better, than VS shut down. But I'm not 100% convinced, and this was much easier.